### PR TITLE
add_hostpath

### PR DESCRIPTION
**Pull request by linclaus**
Add a way to mount hostPath for Prometheus and Alertmanager

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM quay.io/prometheus/busybox:latest
+FROM golang:1.13
```
Comment (coderanger) [inline Dockerfile:1]: Why did you change this?

File: pkg/prometheus/statefulset.go [hunk 2]
```diff
@@ -844,6 +852,8 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 		prometheusConfigReloaderResources.Limits[v1.ResourceMemory] = resource.MustParse(c.ConfigReloaderMemory)
 	}
 
+	privileged := true
```
Comment (coderanger) [inline pkg/prometheus/statefulset.go:855]: Uhhhhhhh, this is a very very bad thing.

**Review by coderanger — COMMENTED** [review]
I think you may have accidentally included a lot more than you meant to. Are you sure you can't do this via the existing `containers` deep merge override?

**Comment by linclaus** [comment]
> 
> 
> I think you may have accidentally included a lot more than you meant to. Are you sure you can't do this via the existing `containers` deep merge override?

**Comment by linclaus** [comment]
> 
> 
> I think you may have accidentally included a lot more than you meant to. Are you sure you can't do this via the existing `containers` deep merge override?

There is something wrong with my local branch, and I will redo it after my correction.
